src/widgets/text_input/crossterm_interactive.rs

```diff
@@ -0,0 +1,315 @@
+            .collect();
+
+        if scan_backwards {
+            char_pairs = char_pairs
+                .into_iter()
+                .take(self.cursor_pos.saturating_sub(1))
+                .rev()
+                .collect();
+        } else {
+            char_pairs = char_pairs.into_iter().skip(self.cursor_pos).collect()
+        }
+
+        if let Some((idx, _chars)) = char_pairs.iter().find(|(_, chars)| {
+            // find a boundary where we go from non-whitespace to whitespace
+            match (chars[0].is_whitespace(), chars[1].is_whitespace()) {
+                (true, true) => false,
+                (true, false) => scan_backwards,
+                (false, true) => !scan_backwards,
+                (false, false) => false,
+            }
+        }) {
+            // println!("bounry at {}: '{}{}'", idx, _chars[0], _chars[1]);
+            if scan_backwards {
+                idx + 1
+            } else {
+                idx + 2
+            }
+        } else {
+            // no whitespace boundary found, remove to start of string
+            if scan_backwards {
+                0
+            } else {
+                self.value.len()
+            }
+        }
+    }
+
+    fn handle_modifiers(&mut self, modifiers: KeyModifiers, code: KeyCode) -> InteractionOutcome {
+        match (modifiers, code) {
+            // delete to current word start
+            (KeyModifiers::CONTROL, KeyCode::Char('w')) => {
+                // find the first boundary going from non-whitespace to whitespace,
+                // going backwards from the cursor position
+                // println!("up to cursor ({}): '{}'", self.cursor_pos, self.up_to_cursor());
+
+                let remove_to = self.cursor_pos as usize;
+                let remove_from = self.word_boundary_idx_under_cursor(true);
+
+                // println!("removing span '{}'", &self.value.as_str()[remove_from..remove_to]);
+
+                // and collect everything that isn't between [remove_from..remove_to)
+                self.cursor_pos = remove_from;
+                self.value = self
+                    .value
+                    .chars()
+                    .take(remove_from)
+                    .chain(self.value.chars().skip(remove_to))
+                    .collect();
+            }
+            // jump to end of line
+            (KeyModifiers::CONTROL, KeyCode::Char('e')) => {
+                self.cursor_pos = self.value.len();
+            }
+            // jump to start of line
+            (KeyModifiers::CONTROL, KeyCode::Char('a')) => {
+                self.cursor_pos = 0;
+            }
+            // jump back a word
+            (KeyModifiers::ALT, KeyCode::Char('b')) => {
+                self.cursor_pos = self.word_boundary_idx_under_cursor(true);
+            }
+            // jump forward a word
+            (KeyModifiers::ALT, KeyCode::Char('f')) => {
+                self.cursor_pos = self.word_boundary_idx_under_cursor(false);
```

Comment:
Are these going to be changed to match their non-modifier counterparts (and to be consistent with most terminals?)